src/dbally/collection/collection.py

```diff
@@ -192,49 +226,69 @@ async def ask(
 
         # select view
         views = self.list()
-
-        if len(views) == 0:
-            raise ValueError("Empty collection")
-        if len(views) == 1:
-            selected_view = next(iter(views))
-        else:
-            selected_view = await self._view_selector.select_view(
-                question=question,
-                views=views,
+        selected_view = None
+        result = None
+
+        try:
```

Comment:
It's getting messy here. Maybe we should add a utils decorator to bind exceptions to handlers and move the chain execution to a separate method. I'm thinking of something like this:
```python
@handle_exception([NoViewFoundError,...], self.ask_fallback)
def ask(self, ...):
```